src/ir/ty.rs

```diff
@@ -365,6 +353,16 @@ impl IsOpaque for Type {
     }
 }
 
+/// Whether this type has a vtable.
+impl HasVtable for Type {
+    // Note that we need the Type's Item to answer this question.
+    type Extra = ItemId;
+
+    fn has_vtable(&self, ctx: &BindgenContext, itemid: &ItemId) -> bool {
+        ctx.lookup_item_id_has_vtable(itemid)
+    }
```

Comment:
Right -- anyone calling this impl *could* (doesn't mean that they are now, but they could) just call the `Item` or `ItemId` impl and it would be cleaner than the hoops being jumped through here.

Yeah, I've convinced myself: let's remove the `HasVtable` impls for everything except `Item` and `ItemId`.